compiler/rustc_passes/src/errors.rs

```diff
@@ -864,12 +864,11 @@ impl IntoDiagnostic<'_> for InvalidAttrAtCrateLevel {
         diag.set_arg("name", self.name);
         // Only emit an error with a suggestion if we can create a string out
         // of the attribute span
-        if let Some(src) = self.snippet {
-            let replacement = src.replace("#!", "#");
+        if let Some(src) = self.snippet && src.starts_with("#![") {
             diag.span_suggestion_verbose(
-                self.span,
+                self.span.with_lo(self.span.lo() + BytePos(1)).with_hi(self.span.lo() + BytePos(2)),
```

Comment:
`snippet` in `InvalidAttrAtCrateLevel` is no longer used, this span can be stored there instead.